lms/static/js/ccx/view/schedule_right_container.js

```diff
@@ -0,0 +1,221 @@
+          startDate,
+          dueDate
+        );
+      },
+
+      showAllUnitsInScheduleTree: function(e) {
+        // add unit to schedule tree.
+        e.preventDefault();
+        this.trigger(
+          'showAllUnitsInScheduleTree'
+        );
+      },
+
+      onChapterSelect: function (e) {
+        // On chapter select populate subsection drop box.
+        var $chapterSelect = $(e.currentTarget);
+        var chapterLocation = $chapterSelect.val();
+        var $sequentialSelect = this.$el.find('#ccx_sequential');
+        if (chapterLocation !== 'none') {
+          this.selectedChapter = this.collection.findUnit(
+            this.chapters, chapterLocation
+          );
+          this.enableAddUnitToScheduleButton = true;
+          $sequentialSelect.html('');
+          this.render();
+        } else {
+          this.resetFormFlags();
+          this.render();
+        }
+      },
+
+      onSubsectionSelect: function(e) {
+        // On subsection aka sequential select populate verticals drop box.
+        var $subSectionSelect = $(e.currentTarget);
+        var subSectionLocation = $subSectionSelect.val();
+        var $verticalSelect = this.$el.find('#ccx_vertical');
+
+        if (subSectionLocation !== 'all') {
+          this.selectedSubsection = this.collection.findUnit(
+            this.chapters,
+            this.selectedChapter.location,
+            subSectionLocation
+          );
+          $verticalSelect.html('');
+          this.render();
+        } else {
+          this.resetSubsectionSelectFlags();
+          this.render();
+        }
+      },
+
+      onVerticalSelect: function(e) {
+        // On vertical select disable edit date sections.
+        var $verticalSelect = $(e.currentTarget);
+        var verticalLocation = $verticalSelect.val();
+
+        if (verticalLocation !== 'all') {
+          this.disableFields(this.$('.ccx_start_date_time_fields'));
+          this.disableFields(this.$('.ccx_due_date_time_fields'));
+        } else {
+          this.enableFields(this.$('.ccx_start_date_time_fields'));
+          this.enableFields(this.$('.ccx_due_date_time_fields'));
+        }
+      },
+
+      saveSchedule: function(e) {
+        // save schedule on server.
+        e.preventDefault();
+        var $button = $(e.currentTarget);
+        $button.prop('disabled', true).text(gettext("Saving"));
+        this.trigger(
+          'saveSchedule'
+        );
+      },
+
+      resetSaveChangesButton: function() {
+        // After collection save enable save changes button
+        this.$el.find('#save-changes').prop(
+          'disabled', false
+        ).text(gettext("Save changes"));
+      },
+
+      onDateSelect: function(e) {
+        // on date select from date picker format it.
+        var $date = $(e.currentTarget);
+        var date = $date.datepicker( "getDate" );
+
+        if (date) {
+          $(this).val(date.getFullYear() + "-" +
```

Comment:
Can you comment this code? I'm surprised that there isn't a cleaner way to get formatted date information from a date picker.

@efischer19 You've done similar work recently. Do you have any suggestions?